engines/agos/event.cpp

```diff
@@ -459,6 +459,22 @@ void AGOSEngine::delay(uint amount) {
 
 		while (_eventMan->pollEvent(event)) {
 			switch (event.type) {
+			case Common::EVENT_JOYBUTTON_DOWN:
+				_joyaction = event.joystick;
+			case Common::EVENT_CUSTOM_ENGINE_ACTION_START:
+				_action = (AGOSAction)event.customType;
+
+				if (event.customType == AGOSAction::kActionToggleFastMode) {
+					_fastMode = !_fastMode;
+				}
+				break;
+			case Common::EVENT_JOYBUTTON_UP:
+				_joyaction.axis = 0;
+				_joyaction.button = 0;
+				_joyaction.position = 0;
```

Comment:
Minor nitpick - I'd prefer to keep the  EVENT_JOYBUTTON_DOWN & EVENT_JOYBUTTON_UP closer together.